plugins/installDatabase/InstallSchema.civi-setup.php

```diff
@@ -76,11 +80,10 @@ public function installDatabase(\Civi\Setup\Event\InstallDatabaseEvent $e) {
     $sqlPath = $model->srcPath . DIRECTORY_SEPARATOR . 'sql';
     $spec = $this->loadSpecification($model->srcPath);
 
-    \Civi\Setup::log()->info(sprintf('[%s] generateCreateSql', basename(__FILE__)));
-    \Civi\Setup\DbUtil::sourceSQL($model->db, \Civi\Setup\DbUtil::generateCreateSql($model->srcPath, $spec->database, $spec->tables));
+    require_once $model->settingsPath;
 
-    \Civi\Setup::log()->info(sprintf('[%s] generateNavigation', basename(__FILE__)));
-    \Civi\Setup\DbUtil::sourceSQL($model->db, \Civi\Setup\DbUtil::generateNavigation($model->srcPath));
+    \Civi\Setup::log()->info(sprintf('[%s] generateCreateSql', basename(__FILE__)));
+    \Civi\Setup\DbUtil::sourceSQL($model->db, \Civi\Setup\DbUtil::generateCreateSql($model->srcPath, $model->db['database'], $spec->tables));
```

Comment:
Hrm, I'm confused -- this seems to be flip-floppy.

 1. Earlier commit called `generateCreateSql($model->srcPath, $model->db['database'], ...)`. (This passes in a `$database` as the DB name, a string.)
 2. Later commit called `generateCreateSql($model->srcPath, $spec->database, ...)` (This is in a `$database` as an array-data-structure.)
 3. Even later commit flipped back to `generateCreateSql($model->srcPath, $model->db['database'], ...)`

I feel the second notation is correct:

 * The Smarty template `schema.tpl` uses `$database` like an array/object (e.g. `{$database.comments}`. The first and third revisions treat it like a string.
 * In the original GenCode context, the *actual target database-name is not known or knowable*. There's no way those templates could have meaningfully used the string DB name.
 * When the older `CRM/Core/CodeGen/Schema.php` loads the Smarty template `schema.tpl`, it sets `database` using `$this->config->database`. Tracing this backward, that data is an array produced by `CRM_Core_CodeGen_Specification::getDatabase()`.